hooks/pre/30-upgrade.rb

```diff
@@ -80,6 +79,30 @@ def fix_katello_settings_file
   end
 end
 
+def mongo_mmapv1_check
+  custom_hiera = '/etc/foreman-installer/custom-hiera.yaml'
+  mongodb_dir = '/var/lib/mongodb/'
+  # check and see if we have a pulp_database already, we have a wiredTiger file, and if the Hiera file has already been modifed.
+  if File.file?("#{mongodb_dir}/pulp_database.0") && File.file?("#{mongodb_dir}/WiredTiger.wt")
+    if File.foreach("#{custom_hiera}").grep(/mongodb::server::storage_engine:/).any?
+      logger.info 'No changed needed, Mongo storage engine will installed with wiredTiger'
+    else
+      # set storage engine to MMAPv1 in Hiera file and create engine file.
+      logger.info 'Detecting Pulp database and no wiredTiger files, keeping storage engine as MMAPv1'
+      logger.info 'To upgrade to wiredTiger at a later time run foreman-installer with the --upgrade-mongo-storage flag.'
+      # Write to custom_hiera to tell users to not touch the setting.
+      open(custom_hiera, 'a') { |f|
+        f << "# Added by foreman-installer during upgrade, do not remove or Mongo will not start!!!!\n"
```

Comment:
I think the exclamation points are slightly excessive. Perhaps it should instruct the user about the ```upgrade-mongo-storage``` here as well since that's the proper fix.